src/ripple/overlay/impl/PeerImp.cpp

```diff
@@ -2309,6 +2313,23 @@ PeerImp::onMessage(std::shared_ptr<protocol::TMGetObjectByHash> const& m)
     }
 }
 
+void
+PeerImp::onMessage(std::shared_ptr<protocol::TMSquelch> const& m)
+{
+    auto validator = m->validatorpubkey();
+    PublicKey key(Slice(validator.data(), validator.size()));
```

Comment:
Constructing `key` may throw if `validator` has bad data. Please see this example:
https://github.com/ripple/rippled/blob/2d79e593ee1dc54092d25ffe9d3d461037f76965/src/ripple/overlay/impl/PeerImp.cpp#L1237